apps/dashboard/src/@3rdweb-sdk/react/hooks/useBalanceForAddress.tsx

```diff
@@ -1,13 +1,23 @@
+import { thirdwebClient } from "@/constants/client";
 import { useQuery } from "@tanstack/react-query";
-import { useSDK } from "@thirdweb-dev/react";
+import { defineChain } from "thirdweb";
+import { getWalletBalance } from "thirdweb/wallets";
 import invariant from "tiny-invariant";
 import { walletKeys } from "../cache-keys";
 
-export function useBalanceForAddress(address: string | undefined) {
-  const sdk = useSDK();
+export function useBalanceForAddress(
+  address: string | undefined,
+  chainId: number | undefined,
+) {
   invariant(address, "Address is not provided");
+  invariant(chainId, "chainId is undefined");
   return useQuery(walletKeys.balances(address), async () => {
```

Comment:
since `chainId` is part of the logic it also should be part of the queryKey, feel free to define it here directly or update the `walletKeys.balances()`